CMakeLists.txt

```diff
@@ -167,6 +182,7 @@ set(DYNAMIC_LIB
     ${THRIFTNB_LIB}
     ${OPENSSL_LIBRARIES}
     ${OPENSSL_CRYPTO_LIBRARY}
+    ${MESALINK_LIB}
```

Comment:
What if WITH_MESALINK is defined but openssl is in system and still be found? In the current cmake, brpc may still use openssl?

Reply:
MesaLink provides SSL/TLS but not crypto; so OpenSSL's libcrypto is still linked.

Actually, either `${OPENSSL_LIBRARIES}` or  `${OPENSSL_SSL_LIBRARY}` is just redundant in the cases of MesaLink builds. No code uses libssl; therefore the generated libbrpc.so does not link libssl. 

```shell
ldd output/lib/libbrpc.so
```
yields the following result:
https://paste.ubuntu.com/p/bCMkRJyFK9/

So you can see there's no libssl linked.

Reply:
If you think this is not preferred for readability, I have a quickfix using CMake's `list(APPEND)`. Looking forward to your reply.

Reply:
Thanks for your detailed explanation. In any case either ${OPENSSL_LIBRARIES} or ${MESALINK_LIB} is used,  it's enough to link just one of them. So it may be more clear if we comment ${OPENSSL_LIBRARIES} when mesalink is used.